src/main/scala/TypeClass.scala

```diff
@@ -0,0 +1,46 @@
+trait Semigroup[A] {
+  def plus(a: A, b: A): A
+}
+object Monoid {
+  // If we have a Monoid[A], we also have a Semigroup[A]
+  implicit def fromCanAddZ[A](implicit canAddZ: CanAddZ[A]): Semigroup[A] = canAddZ
+  def apply[A](implicit canAdd: Semigroup[A]): Semigroup[A] = canAdd
+}
+
+trait CanAddZ[A] extends Semigroup[A]{
+  def zero: A
+}
+object CanAddZ {
+  implicit val canAddInts: CanAddZ[Int] = new CanAddZ[Int] {
```

Comment:
and rename to `monoidInt` and similar below

Reply:
we found a better name you poop